lib/cli/interactive-setup/console-dev-mode-feed.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const { writeText, style } = require('@serverless/utils/log');
+const WebSocket = require('ws');
+const chalk = require('chalk');
+const { devModeFeed } = require('@serverless/utils/lib/auth/urls');
+const consoleUi = require('@serverless/utils/console-ui');
+const resolveAuthMode = require('@serverless/utils/auth/resolve-mode');
+const streamBuffers = require('stream-buffers');
+const apiRequest = require('@serverless/utils/api-request');
+const promptWithHistory = require('@serverless/utils/inquirer/prompt-with-history');
+
+const streamBuff = new streamBuffers.ReadableStreamBuffer({
+  frequency: 500,
+  chunkSize: 2048 * 1000000,
+});
+
+const consoleMonitoringCounter = {
+  logBatches: 0,
+  events: 0,
+  responses: 0,
+};
+
+const handleSocketMessage = (data) => {
+  try {
+    const splitData = data.toString('utf-8').split(';;;');
+    const jsonArray = splitData.reduce((arr, item) => {
+      try {
+        const parsedItem = JSON.parse(item);
+        if (Array.isArray(parsedItem)) {
+          return [...arr, ...parsedItem];
+        }
+        throw new Error('Not an array');
+      } catch (error) {
+        return arr;
+      }
+    }, []);
+    if (!Array.isArray(jsonArray)) return;
+    const sortedItems = consoleUi.omitAndSortDevModeActivity(jsonArray);
+
+    for (const activity of sortedItems) {
+      const resourceName = ((activity.tags || {}).aws || {}).resourceName;
+      const time = consoleUi.formatConsoleDate(new Date(activity.timestamp));
+
+      switch (activity.type) {
+        case 'log':
+          consoleMonitoringCounter.logBatches += 1;
+          process.stdout.write(chalk.grey(`${time} • ${resourceName} • Log\n`));
+          try {
+            const parsedBody = JSON.parse(activity.body);
+            if (typeof parsedBody === 'string') {
+              throw new Error('Not a JSON object');
+            }
```

Comment:
Yes. In the events that a customer sends a string response from their lambda function, for example `"Foo bar"`, this will parse correctly in the `JSON.parse` function and be returned correctly from the customers lambda function but will not be a JSON object 👍

Reply:
Ok, so wouldn't be more natural to write following:

```javascript
try {
  if (noObject) throw;
  // object case
} case (error)
 // string case
}
```

as:

```javascript
if (noObject) // string case
else // object case
```

So without try/catch handling?

Reply:
This would not work. We need the try/catch because we want to pretty print any logs that are formatted as json but logs are not always JSON objects so if the log message was a simple string or a non json object this would throw an error.